compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10529,6 +10527,19 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
          cg->decReferenceCount(source2Node);
          }
 
+      if (baseSource1Ref->getOffset() == baseSource2Ref->getOffset())
```

Comment:
Not sure if offsets are expected but later in original code, we check those therefore I decided to check as well.